File: express/src/service/UserService.ts
import { User } from '../entity/User';

export interface iUserService {
  findByPageAndSizeAndFilterAndOrder(page: number, size: number, filter?: string, order?: string, ascendingDirection?: boolean): Promise<[User[], number]>;
  findOne(id: number): Promise<User>;
  update(id: number, entity: User): Promise<boolean>;
  remove(id: number): Promise<number>;
}

File: spring/src/main/java/hu/marko/szakdolgozat/spring/service/UserService.java
package hu.marko.szakdolgozat.spring.service;

import hu.marko.szakdolgozat.spring.service.model.PageModel;
import hu.marko.szakdolgozat.spring.service.model.User;

public interface UserService {
  PageModel<User> findByPageAndSizeAndFilterAndOrder(Integer page, Integer size, String filter, String order,
      Boolean ascendingDirection);

  User findOne(Long id);

  Boolean update(Long id, User user);

  Long remove(Long id);
}
